### DISCUSSION TITLE
Need to evaluate the library for use with React Native based app

### DISCUSSION THREAD

**Discussion by jnmayur** [xmppjs/xmpp.js#950] [General]
Hi, we're moving from a web view based implementation to a React Native based implementation.
We've tried using the xmppp.js package as is in our React Native app and it just works fine. We'd want to read more and understand if this is the right way to build a chat app or we should use a native client and use bridges instead.
Also, is there any benchmarking already done in terms of performance?

**Comment by pcsaunak** [comment]
Hi, I am trying to implement the xmpp.js package in my react native app but I am unable to do so. I keep getting an error saying 
`Error: ENOENT: no such file or directory, open '/Users/saunak/Documents/Personal/POC/XMPP_POC/xmpp_client_js/http:/192.168.68.102:19000/node_modules/expo/AppEntry.bundle?platform=ios&dev=true&hot=false'`

Is there any sample that I can refer to , I have been trying to use the sample that has been shown in the xmpp/client pacakge

**Reply by pcsaunak** [reply]
I was able to connect to my server but if I check my server logs I see the below error
`2022-06-21 07:55:46.688 [warning] <0.528.0>@mod_mqtt_session:log_disconnection:1362 Rejected MQTT connection from ::ffff:1.1.1.1: Protocol error: Unexpected PUBLISH packet`

**Reply by pcsaunak** [reply]
The log message from the client is 
`<message to="sss@example.com" type="chat" xmlns="jabber:client"><body>Hello</body></message>`

**Comment by pcsaunak** [comment]
So I have made changes to the service URI and now I am able to send messages however I am not able to receive messages. 
I am really stuck here and looking for help.

**Reply by sonnyp** [reply]
Send your presence.

https://github.com/xmppjs/xmpp.js/tree/main/packages/client#example
